cbrkit/sim/collections.py

```diff
@@ -190,3 +192,227 @@ def wrapped_func(query: Sequence[ValueType], case: Sequence[ValueType]) -> float
+    list_weights: List[Weight] = None,
+    exact: bool = False,
+    return_local_similarities: bool = True,
+) -> Callable[[Collection[Any], Collection[Any]], ListMappingSim]:
+    """List Mapping Weighted similarity function.
+
+    Parameters:
+    element_similarity (callable): The similarity function to use for comparing elements.
+    list_weights (list): The list of weights to use for different similarity intervals.
+    exact (bool): The comparison type, either exact or inexact. Default is False (inexact).
+    return_local_similarities (bool): Whether to return local similarities. Default is True.
+
+    Examples:
+        >>> weights = [Weight(weight=1.0, lower_bound=0.0, upper_bound=0.1, inclusive_lower=True, inclusive_upper=True)]
+        >>> sim = list_mapping_weighted(lambda x, y: 1.0 if x == y else 0.0, weights)
+        >>> result = sim(["a", "b", "cd"], ["a", "b", "c"])
+        >>> result.value
+        0.0
+        >>> result.local_similarities
+        [1.0, 1.0, 0.0]
+    """
+
+    if list_weights is None:
+        list_weights = []
+
+    list_mapping_func = list_mapping(
+        element_similarity, exact, return_local_similarities
+    )
+
+    def wrapped_func(x: Collection[Any], y: Collection[Any]) -> ListMappingSim:
+        result = list_mapping_func(x, y)
+
+        if return_local_similarities:
+            similarity = result.value
+            local_similarities = result.local_similarities
+        else:
+            similarity = result.value
+            local_similarities = []
+
+        total_weighted_sim = 0.0
+        total_weight = 0.0
+
+        # Arrange and normalize weights
+        for weight in list_weights:
+            weight_range = weight.upper_bound - weight.lower_bound
+            weight.normalized_weight = weight.weight / weight_range
+
+        for sim in local_similarities:
+            for weight in list_weights:
+                lower_bound = weight.lower_bound
+                upper_bound = weight.upper_bound
+                inclusive_lower = weight.inclusive_lower
+                inclusive_upper = weight.inclusive_upper
+
+                if (
+                    (inclusive_lower and lower_bound <= sim <= upper_bound)
+                    or (not inclusive_lower and lower_bound < sim <= upper_bound)
+                ) and (inclusive_upper or sim < upper_bound):
+                    weighted_sim = weight.normalized_weight * sim
+                    total_weighted_sim += weighted_sim
+                    total_weight += weight.normalized_weight
+
+        if total_weight > 0:
+            final_similarity = total_weighted_sim / total_weight
+        else:
+            final_similarity = similarity
+
+        return ListMappingSim(
+            value=final_similarity,
+            local_similarities=local_similarities
+            if return_local_similarities
+            else None,
+        )
+
+    return wrapped_func
+
+
+def list_correctness(
+    discordant_parameter: float = 1.0,
```

Comment:
The naming is a bit odd. What about `worst_case_sim` or `fallback_sim`.

Also, is a default value of `1.0` sensible here? Given the description, I would assume that `0.0` could also make sense.